ctd/ctd.py

```diff
@@ -352,8 +352,8 @@ def from_btl(fname, lon=None, lat=None):
     datetimes = dates + ' ' + times
 
     # Fill the Date column with datetimes.
-    cast['Date'].iloc[::4] = datetimes.values
-    cast['Date'].iloc[1::4] = datetimes.values
+    cast['Date'].iloc[::len(rowtypes)] = datetimes.values
+    cast['Date'].iloc[1::len(rowtypes)] = datetimes.values
```

Comment:
:+1: if you want this merged ASAP please just send a PR with only this change while we discuss the changes in mpl later.

Reply:
Done!